### PR TITLE
Views

### PR DESCRIPTION
**Pull request by Rafiduzzaman**
**Unveiling the Next Chapter: Transformative Blog App Project**

Embark on a journey through innovation and functionality with our latest project – the Blog App! We are thrilled to present a sneak peek into the wireframes that will shape a seamless and engaging user experience.

### Project Overview

In this ambitious venture, our primary goal is to implement the design from the captivating sneak peek wireframes. While we acknowledge the importance of styling, our initial focus lies in ensuring that every element functions seamlessly. Please note that login logic will be addressed in the upcoming milestone through the integration of Devise.

### Wireframes Preview

Let's dive into the intricate details of our wireframes, meticulously designed to enhance user navigation and interaction:

1. **All Users Page Wireframe**
   image
2. **Single User Page Wireframe**
   image

3. **All Posts by a Given User Page Wireframe**
   image

4. **Single Post Page Wireframe**
   image

### CODE REVIEW

File: app/views/partials/_user_profile.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  <div class='mb-4 flex gap-2 items-center border-2 border-pink-900 rounded-2xl overflow-hidden shadow'>
+    <div class='m-3.5'>
+      <%= image_tag user.photo_url, alt: "profil image", width: "100px", height: "100px" %>
+    </div>
+    <div class='relative w-full'>
+      <p class='text-3xl text-pink-900 font-bold'>
+          <%= user.name %>
+      </p>
+      <span class='text-sm text-pink-500 absolute bottom-0 right-4'>
+        Number of posts: <%= user.posts_counter%>
```
Comment (franclobo) [inline app/views/partials/_user_profile.html.erb:11]: - [x] The post counter is not working I recommend adding this line to your post.rb model 👉🏽 `after_create :update_user_posts_counter`

| Before | After |
|---|---|
| image | image |

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,2 +1,22 @@
...
+        <ul class='flex flex-col gap-2'>
+            <% @user.three_most_recent_posts.each_with_index do |post, index| %>
+                <li>
+                    <%= render partial: 'partials/post_item', locals: { post: post, index:index } %>
+                </li>
+            <% end %>
+        </ul>
+    </div>
+    <div>
+        <%= link_to 'See all posts', user_posts_path(@user.id), method: :get, class: 'bg-gray-700 text-white py-2 px-6 rounded'%>
```
Comment (franclobo) [inline app/views/users/show.html.erb]: - [ ] When clicking on this button an error arose, please 🙏🏽 fix it to be able to see all posts 👇🏽 
image

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Rafiduzzaman, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Good use of partials ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @Rafiduzzaman 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Setup linters correctly ✔️ 
- Created **_index and show pages_** for the `User` model ✔️ 
- Created **_index and show pages_** for the `Post` model ✔️ 
- Great use of partials ✔️ 
- Great use of custom methods to display certain features ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**